sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/ProcessEventArgs.cs

```diff
@@ -27,7 +27,7 @@ public struct ProcessEventArgs
         ///
         /// <value><c>true</c> if the arguments contain an event to be processed; otherwise, <c>false</c>.</value>
         ///
-        public bool HasEvent => ((Data == null) || (Partition == null));
+        public bool HasEvent => ((Data != null) && (Partition != null));
```

Comment:
It started life as `IsEmpty`.   Apparently, I changed the name and semantics but not the actual logic.    I left the constructor and check for the partition here as-is, since the "empty partition context" concept was something that we added specific to this instance of the event processor.   If we decide to make that more widely available, we should adjust as you're suggesting.

I'm going to defer that until we look closer at how we would reintroduce the concept of a base class.    I have a feeling that we'll end up either allowing empty construction directly or moving to the Shared project to support `internal` for it.